src/@next/components/organisms/AdyenPaymentGateway/AdyenPaymentGateway.tsx

```diff
@@ -0,0 +1,183 @@
+   */
+  processPayment: () => void;
+  /**
+   * Method to call on gateway payment submission.
+   */
+  submitPayment: (data: {
+    confirmationData: any;
+    confirmationNeeded: boolean;
+  }) => Promise<any>;
+  /**
+   * Method called after succesful gateway payment submission. This is the case when no confirmation is needed.
+   */
+  submitPaymentSuccess: (
+    order?: CompleteCheckout_checkoutComplete_order
+  ) => void;
+  /**
+   * Errors returned by the payment gateway.
+   */
+  errors?: IFormError[];
+  /**
+   * Method called when gateway error occured.
+   */
+  onError: (errors: IFormError[]) => void;
+}
+
+const AdyenPaymentGateway: React.FC<IProps> = ({
+  config,
+  formRef,
+  scriptConfig,
+  styleConfig,
+  processPayment,
+  submitPayment,
+  submitPaymentSuccess,
+  errors,
+  onError,
+}: IProps) => {
+  const adyenClientKey = config?.find(({ field }) => field === "client_key")
+    ?.value;
+  const adyenConfig = config?.find(({ field }) => field === "config")?.value;
+  const parsedAdyenConfig = adyenConfig && JSON.parse(adyenConfig);
+
+  const [dropin, setDropin] = useState<any>();
+  const gatewayRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (adyenClientKey && parsedAdyenConfig && !dropin && gatewayRef.current) {
+      const link = document.createElement("link");
+      link.rel = "stylesheet";
+      link.href = styleConfig.src;
+      link.integrity = styleConfig.integrity;
+      link.crossOrigin = styleConfig.crossOrigin;
+      document.body.appendChild(link);
+
+      const script = document.createElement("script");
+      script.src = scriptConfig.src;
+      script.integrity = scriptConfig.integrity;
+      script.crossOrigin = scriptConfig.crossOrigin;
+      script.async = true;
+      script.onload = initAdyenGatewayHandlers; // Wait until the script is loaded before initiating AdyenCheckout
+      document.body.appendChild(script);
+    }
+  }, [adyenClientKey, parsedAdyenConfig, gatewayRef.current]);
+
+  const initAdyenGatewayHandlers = () => {
+    const configuration = adyenClientKey &&
+      adyenConfig && {
+        locale: navigator.language,
+        environment: "test",
+        clientKey: adyenClientKey,
+        paymentMethodsResponse: parsedAdyenConfig,
+        showPayButton: false,
+        onSubmit: (state: any, dropin: any) => {
+          if (!state?.isValid) {
+            onError([new Error("Invalid payment submission")]);
+          } else {
+            submitPayment(state?.data)
+              .then(value => {
+                if (value.error) {
+                  onError([value.error]);
+                } else if (!value?.confirmationNeeded) {
+                  submitPaymentSuccess(value?.order);
+                } else {
+                  const paymentAction =
+                    value?.confirmationData &&
+                    JSON.parse(value?.confirmationData);
+                  dropin.handleAction(paymentAction);
```

Comment:
I'd improve code readability here
Also I don't think `undefined` is legit `dropin.handleAction` argument, please correct me if I'm wrong

```suggestion
                  if(value?.confirmationData) {
                    const paymentAction = JSON.parse(value?.confirmationData);
                    dropin.handleAction(paymentAction)
                  }
                  const paymentAction =
                    value?.confirmationData &&
                    JSON.parse(value?.confirmationData);
                  dropin.handleAction(paymentAction);
```

Reply:
good point!